joystickwake: init at 0.2.4

bertof opened the pull request:
<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

###### Motivation for this change
Add package for [joystickwake](https://github.com/foresto/joystickwake), an utility for disabling screensavers and auto-lock while any input is received from a gamepad. This solves the annoying issue of auto-lock locking the session while playing games.

###### Things done
- Add package definition in `pkgs/tools/games/joystickwake/default.nix`
- Add package to the `all-packages.nix` file
- Add myself as maintainer

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

r-rmcgibbo commented:
Result of `nixpkgs-review pr 127879` at 949844cf run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>joystickwake</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/no-python-tests.md">no-python-tests</a>

Test runner could not discover any test cases: ‘Ran 0 tests in 0.000s’
Near pkgs/tools/games/joystickwake/default.nix:20:0:
```
   |
20 |     description = "A joystick-aware screen waker";
   | ^
```
</li>
  </ul>
</details>

SuperSandro2000 commented on maintainers/maintainer-list.nix:
```diff
@@ -1220,6 +1220,16 @@
     github = "benneti";
     githubId = 11725645;
   };
+  bertof = {
```
Please put this into a separate commit with the message ``maintainers: add NAME`` and sort it alphabetically.

bertof replied:
Ok

SuperSandro2000 commented on pkgs/tools/games/joystickwake/default.nix:
```diff
@@ -0,0 +1,31 @@
+{ lib, python3, fetchFromGitHub, withX11 ? true }:
+python3.pkgs.buildPythonApplication rec {
```
```suggestion
{ lib, python3, fetchFromGitHub }:

python3.pkgs.buildPythonApplication rec {
```

SuperSandro2000 commented on pkgs/tools/games/joystickwake/default.nix:
```diff
@@ -0,0 +1,31 @@
+    sha256 = "0j8xwfmzzmc9s88zvzc3lv67821r6x28vy6vli3srvx859wprppd";
+  };
+
+  buildInputs = with python3.pkgs;
+    ([ pyudev ] ++ (lib.optional withX11 xlib));
+
```
```suggestion
```

SuperSandro2000 commented on pkgs/tools/games/joystickwake/default.nix:
```diff
@@ -0,0 +1,31 @@
+  buildInputs = with python3.pkgs;
+    ([ pyudev ] ++ (lib.optional withX11 xlib));
+
+  propagatedBuildInputs = with python3.pkgs;
+    ([ pyudev ] ++ (lib.optional withX11 xlib));
```
```suggestion
  propagatedBuildInputs = with python3.pkgs; [ pyudev xlib ];
```

SuperSandro2000 commented on pkgs/tools/games/joystickwake/default.nix:
```diff
@@ -0,0 +1,31 @@
+    maintainers = with maintainers; [ bertof ];
+    license = licenses.mit;
+    platforms = platforms.linux;
+  };
+
+}
```
```suggestion
  };
}
```

bertof committed:
maintainers: add bertof

bertof committed:
joystickwake: init at 0.2.4

Add package for [joystickwake](https://github.com/foresto/joystickwake), an utility for disabling screensavers and auto-lock while any input is received from a gamepad. This solves the annoying issue of auto-lock locking the session while playing games.

bertof commented:
I applied all the proposed changes

bertof commented:
Just a question. From the application readme `xlib` is optional and this may also work on Wayland. Is there any way of indicating it?

SuperSandro2000 commented:
> Just a question. From the application readme `xlib` is optional and this may also work on Wayland. Is there any way of indicating it?

No, not really.
